### PR TITLE
Switch to using the GASNet EX API as the default

### PR DESCRIPTION
**Pull request by jhh67**
Make `comm-gasnet-ex.c` the default GASNet shim instead of `comm-gasnet.c`.  This will facilitate future development using the GASNet-EX API. As a result of this change, the standard `gasnet` tests in `util/cron` will use GASNet-EX, so the `*gasnet-ex*` tests scripts have been renamed to `*gasnet-1*` and modified to use the GASNet-1 shim. Eventually these tests will be removed when the GASNet-1 shim is deprecated.

Resolves https://github.com/Cray/chapel-private/issues/6356.

### CODE REVIEW

**Review by jabraham17 — APPROVED** [review]

**Review by bonachea — APPROVED** [review]
LGTM! Thanks @jhh67!

CC: @PHHargrove

File: util/cron/test-gasnet-1-everything.bash
Comment (bonachea) [inline util/cron/test-gasnet-1-everything.bash:1]: Based on CI failures, it appears `CHPL_NIGHTLY_TEST_CONFIG_NAME` also needs to be updated in each script.

**Comment by jhh67** [comment]
I ran the core benchmarks as described in https://github.com/Cray/chapel-private/issues/3593 with `CHPL_GASNET_VERSION=1` vs `CHPL_GASNET_VERSION=ex` to measure the impact of this change. The benchmarks were compiled as follows:
```
chpl --fast test/studies/prk/Stencil/optimized/stencil-opt.chpl -sorder=sqrt(16e9*numLocales / 8):int
chpl --fast test/release/examples/benchmarks/hpcc/stream.chpl
chpl --fast test/studies/bale/aggregation/ig.chpl -sN=1000000 -suseBlockArr=true
chpl --fast test/release/examples/benchmarks/hpcc/ra.chpl -suseOn=true -sN_U=2**(n-16) -o ra-on
chpl --fast test/release/examples/benchmarks/hpcc/ra.chpl -suseOn=false -sN_U=2**(n-16) -o ra-rmo
chpl --fast test/studies/isx/isx-hand-optimized.chpl -smode=scaling.weakISO -sn=1048576
```

| Benchmark |    GASNet 1 |  GASNet EX |
|----------:|------------:|-----------:|
|    stream |        2822 |       2823 |
|   stencil |     1234434 |    1243977 |
|       isx |       0.668 |      0.659 |
|        ig |        9.95 |      10.08 |
|    ra-rmo | 0.003735020 | 0.00394379 |
|     ra-on |  0.00669717 | 0.00722995 |

These results were collected on an HPE Cray Apollo system with an Infiniband interconnect. All values are throughput so that higher is better, except for `isx` which is elapsed time so lower is better. The GASNet-EX shim has better performance on all benchmarks.

File: util/cron/test-gasnet-1-everything.bash
Comment (jhh67) [reply util/cron/test-gasnet-1-everything.bash:1]: Thanks for catching that.

File: util/cron/test-perf.hpe-apollo-hdr.gasnet-1-ibv.large.bash [hunk 1]
```diff
@@ -9,16 +9,16 @@ export CHPL_TEST_PERF_CONFIG_NAME='16-node-apollo-hdr'
...
 
 export GASNET_PHYSMEM_MAX="0.90"
 
-# Test GASNet EX
-export CHPL_GASNET_VERSION=ex
+# Test GASNet 1
+export CHPL_GASNET_VERSION=1
 
 nightly_args="${nightly_args} -no-buildcheck"
 perf_args="-performance-description gn-ex-ibv-large -numtrials 1 -sync-dir-suffix gex"
```
Comment (jabraham17) [inline util/cron/test-perf.hpe-apollo-hdr.gasnet-1-ibv.large.bash:24]: make sure to update the `-performance-description` strings as well

**Pull request merged**